### ISSUE TITLE
pre-commit install silently fails when running in git-bash

### ISSUE THREAD

**Issue by julie777** [pre-commit/pre-commit#2245]
### describe your issue

I am actually doing dev work on pyscaffold, which uses pre-commit. Problems there led me to track the problem back to pre-commit.
Running on Windows 10 in git-bash the command `precommit install` produces no output, but has a return code of 1.

I tracked the problem to sys.stderr being used without checking that it is not None
It happens at main.py line 335 -          args = parser.parse_args(argv)

### pre-commit --version

2.17.0

### .pre-commit-config.yaml

```yaml
just an empty file
```

### ~/.cache/pre-commit/pre-commit.log (if present)

I tracked it down and found the exception and printed it for you.

Traceback (most recent call last):
  File "C:\Users\julie\juliehome\.venvs\pyscaffold\lib\site-packages\pre_commit\main.py", line 342, in main
    args = parser.parse_args(argv)
  File "C:\Users\julie\AppData\Local\Programs\Python\Python310\lib\argparse.py", line 1825, in parse_args
    args, argv = self.parse_known_args(args, namespace)
  File "C:\Users\julie\AppData\Local\Programs\Python\Python310\lib\argparse.py", line 1858, in parse_known_args
    namespace, args = self._parse_known_args(args, namespace)
  File "C:\Users\julie\AppData\Local\Programs\Python\Python310\lib\argparse.py", line 2070, in _parse_known_args
    stop_index = consume_positionals(start_index)
  File "C:\Users\julie\AppData\Local\Programs\Python\Python310\lib\argparse.py", line 2026, in consume_positionals
    take_action(action, args)
  File "C:\Users\julie\AppData\Local\Programs\Python\Python310\lib\argparse.py", line 1935, in take_action
    action(self, namespace, argument_values, option_string)
  File "C:\Users\julie\AppData\Local\Programs\Python\Python310\lib\argparse.py", line 1213, in __call__
    subnamespace, arg_strings = parser.parse_known_args(arg_strings, None)
  File "C:\Users\julie\AppData\Local\Programs\Python\Python310\lib\argparse.py", line 1858, in parse_known_args
    namespace, args = self._parse_known_args(args, namespace)
  File "C:\Users\julie\AppData\Local\Programs\Python\Python310\lib\argparse.py", line 2092, in _parse_known_args
    self._get_value(action, action.default))
  File "C:\Users\julie\AppData\Local\Programs\Python\Python310\lib\argparse.py", line 2483, in _get_value
    result = type_func(arg_string)
  File "C:\Users\julie\juliehome\.venvs\pyscaffold\lib\site-packages\pre_commit\color.py", line 94, in use_color
    sys.stderr.isatty() and
AttributeError: 'NoneType' object has no attribute 'isatty'

**Comment by asottile** [comment]
this is a bug in python -- I believe fixed in the next patch version

**Comment by asottile** [comment]
actually, if I recall correctly this is due to pip, so you'll have to downgrade pip and then reinstall pre-commit

**Comment by asottile** [comment]
https://github.com/pypa/pip/issues/10875

**Comment by julie777** [comment]
Thanks for pointing me to the pypi pip details. It was informative.

On the other hand, having a problem and just saying wait until something out of our control changes is not a great solution for users. Pyscaffold does pretty much the same thing with colorizing and doesn't have the problem because our implementation is slightly different. We are doing it on stdout instead of stderr, but here it is for reference.

```
def isatty(stream=None):
    """Detect if the given stream/stdout is part of an interactive terminal.

    Args:
        stream: optionally the stream to check

    Returns:
        bool: result of check
    """
    stream = stream or sys.stdout
    print('is tty', stream, hasattr(stream, "isatty"))
    if hasattr(stream, "isatty"):
        return stream.isatty()

    return False
```

A little defensive programming can go a long way in situations like this. 

I had never thought of a missing stderr before. Given that failing with no output for the user is very problematic, I think I am going to start checking and use stdout if my programs have to exit with an error. I only realized what was happening because my shell prompt shows me the return code of the previous command. I image most users would just think it worked and then wonder by their hooks didn't run.

Because I can't wait, I just modified my local copy of pre-commit.

I will be happy to submit a pull request if you like.
